volpiano_display_utilities/text_volpiano_alignment.py

```diff
@@ -6,238 +6,317 @@
+    if not text_section.is_syllabified:
+        logging.debug("Aligning section with unsyllabified text.")
+        unsyllabified_text = text_section.get_syllable(word_num=0, syllable_num=0)
+        # The volpiano section is flattened to a string prior
+        # to alignment with the text. Additional processing on
+        # the volpiano section depends on the volpiano contents:
+        # - If the volpiano section is a barline, we make sure
+        #  that the section terminates wih the proper spacing.
+        # - If the volpiano section is missing music, we make sure
+        #  that the section is properly sized the length of the
+        #  accompanying text.
+        # - If neither, no additional processing occurs.
+        flattened_volpiano = volpiano_section.flatten_to_str()
+        if volpiano_section.is_barline:
+            processed_volpiano = adjust_music_spacing(
+                flattened_volpiano, len(unsyllabified_text), True
+            )
+            comb_section.append((unsyllabified_text, processed_volpiano))
+        elif volpiano_section.is_missing_music:
+            processed_volpiano = adjust_missing_music_spacing(
+                flattened_volpiano, len(unsyllabified_text)
+            )
+            comb_section.append((unsyllabified_text, processed_volpiano))
         else:
-            logging.debug("Text section has more than expected elements. Flattening.")
-            flattened_section = []
-            for word in text_section:
-                flattened_section.extend(word)
-            full_text = "".join(flattened_section)
-            comb_section.append((full_text, volpiano_section))
+            comb_section.append((unsyllabified_text, flattened_volpiano))
     else:
-        vol_words = VOLPIANO_WORD_REGEX.findall(volpiano_section)
-        for txt_word, vol_word in zip_longest(text_section, vol_words, fillvalue="--"):
-            if txt_word == "--":
-                txt_word = [""]
+        # In sections of syllabified text, align text and volpiano
+        # word by word.
+        logging.debug("Aligning words in section with syllabified text.")
+        aligned_words = _zip_and_align(
+            text_section.section,
+            volpiano_section.section,
+            pad_text=[""],
+            pad_volpiano=["----"],
+        )
+        logging.debug("Aligned words: %s", aligned_words)
+        for txt_word, vol_word in aligned_words:
+            # Align each word of text and volpiano in the section
+            # syllable by syllable.
+            logging.debug(
+                "Aligning syllables in word. Text: %s. Volpiano: %s", txt_word, vol_word
+            )
             comb_wrd = _align_word(txt_word, vol_word)
+            logging.debug("Aligned syllables: %s", comb_wrd)
             comb_section.extend(comb_wrd)
     logging.debug("Aligned section: %s", comb_section)
     return comb_section
 
 
+def _add_text_barline(
```

Comment:
Could this be called `_insert_text_barline`? To me, `add` implies that the barline is being appended.

Reply:
and the same for `_add_volpiano_barline` below.

Reply:
Fixed in [9779f0f](https://github.com/DDMAL/volpiano-display-utilities/pull/26/commits/9779f0fe5e6097e45c97bf17d2d66e424234f4c4)